prometheus_alerting_consistency.md

```diff
@@ -0,0 +1,103 @@
+  * There may be exceptions for some broadly scoped alerts, e.g.: `TargetDown`
+* Alerts MUST include a `severity` label indicating the alert's urgency.
+  * Valid severities are: `critical` or `warning` — see below for
+    guidelines on writing alerts of each severity.
+* Alerts MUST include `summary` and `description` annotations.
+  * Think of `summary` as the first line of a commit message, or an email
+    subject line.  It should be brief but informative.  The `description` is the
+    longer, more detailed explanation of the alert.
+* Alerts SHOULD include a `namespace` label indicating the source of the alert.
+  * Many alerts will include this by virtue of the fact that their PromQL
+    expressions result in a namespace label.  Others may require a static
+    namespace label.
+* All alerts MUST include a `sop_url` annotation.
+  * Standard Operating Procedure (SOP) style documentation for resolving alerts is required. (SOP Template Placeholder - https://issues.redhat.com/browse/SIGSRE-55)
+
+## Critical Alerts
+
+Critical alerts are for alerting current and impending disaster situations.  These alerts page an SRE.  The situation should warrant waking someone in the middle of the
+night.
+
+Timeline: ~5 minutes.
+
+Reserve critical level alerts only for reporting conditions that may lead to service unavailability.
+Failures of individual components should not trigger critical level alerts in general, unless they would result in that condition.
+Configure critical level alerts so they fire before the situation becomes irrecoverable.
+
+Example critical alert:
+
+```yaml
+- alert: BackendServiceStuck
+  expr: backend_state != 1
+  for: 10m
+  labels:
+    severity: critical
+  annotations:
+    summary: 'The backend service is stuck in a non-ready state'
+    description: 'The backend service {{ $labels.name }} in the {{ $labels.resource_namespace }} namespace, managed by operator {{ $labels.pod }} has been in a non-ready state for 10 minutes'
+    sop_url: 'https://example.com/backend_service_stuck.asciidoc'
+```
+
+This alert fires if a backend service has *not* been ready for the last 10 minutes.
+This is a clear example of a critical issue that represents a threat to the operability of the service, and likely warrants paging someone.
+The alert has a clear summary and description annotations, and it links to a SOP with information on investigating and resolving the issue.
+
+The group of critical alerts should be small, very well-defined, highly documented, polished and with a high bar set for entry.
+
+## Warning Alerts
+
+The vast majority of alerts should use this severity.
+Issues at the warning level should be addressed in a timely manner, but don't pose an immediate threat to the operation of the service as a whole.
+
+Timeline: ~60 minutes
+
+If your alert does not meet the criteria in "Critical Alerts" above, it belongs to the warning level or lower.
```

Comment:
"or lower" - according to the Style guide, there are only two valid severities - critical and warning

Reply:
Good catch.
Are there use cases for alerting severity outside of critical & warning?
I can't think of any in the scope of running a service.
Maybe some teams use other severity or lack of severity to mean something else e.g. these alerts are ignored and for informational purposes only or trialing a new alert.
I think that would be outside the scope of this guide though.

Reply:
I also can't think of any. We only ever used those two (critical/warning), and we didn't follow any documentation 😂.

Reply:
We have a third level, called "Info". There is no notification added to this one. It's primary purpose is to help us "understand" if something is happening, but that it is not necessarily important for us to know. An example is if a service has hit some sort of quota threshold which might mean it is being throttled. As SREs, we don't necessarily care if a service is being throttled or not, but Support does because they may get a call from a customer (in the case where the customer doesn't understand also).

Reply:
That's fine @david-martin ... I don't think we need to overcomplicate it. And you 100% correct about "other information" that probably shouldn't be in alerts and can be surfaced elsewhere....however, in our experience, there have been cases where knowing XYZ has helped us debug a certain issue, and having those alerts "together" has been very helpful. That being said, I don't think it is necessary to mention that here as it does complicate it.